### DISCUSSION TITLE
Can't preview JPEG-XL images

### DISCUSSION THREAD

**Discussion by d-air1** [sxyazi/yazi#790] [1. Q&A]
I'm using kde's konsole, from what I have seen here:
![image](https://github.com/sxyazi/yazi/assets/40647198/3083e869-649d-4017-bf34-4fe221b99331)
 konsole is mapped to the old kitty protocol.
However, konsole has supported the much better in my opinion sixel protocol for a few years now https://invent.kde.org/utilities/konsole/-/merge_requests/594. You can confirm this using the img2sixel program provided by the libsixel library. I want to force the use of that protocol.

**Comment by SolitudeSF** [comment]
how is it better?

**Reply by d-air1** [reply]
So far when I use img2sixel it seems to be able to handle newer file formats like jpeg-xl. I was assuming this was a protocol error because yazi was using the old kitty protocol.

**Reply by d-air1** [reply]
After some research regarding terminal image protocols. It may not be the best, but seems to be the most widely supported. Whether or not that is related to my issue, I don't know, but at the very least this could still be a good feature request for terminals that support multiple protocols.

**Reply by SolitudeSF** [reply]
image format is unrelated to display protocol, they all just display rgb data (mostly).

**Reply by d-air1** [reply]
Thanks for the clarification.

**Comment by sxyazi** [comment]
> So far when I use img2sixel it seems to be able to handle newer file formats like jpeg-xl. I was assuming this was a protocol error because yazi was using the old kitty protocol.

This is irrelevant and the reason here is the [`image` crate](https://github.com/image-rs/image) used by Yazi currently does not support jpeg-xl.

You can create a custom previewer that calls tools like `imagemagick` to convert jpeg-xl to a regular jpeg for display, just like the PDF previewer does, you can refer to https://github.com/sxyazi/yazi/blob/main/yazi-plugin/preset/plugins/pdf.lua, and replace `pdftoppm` with `imagemagick` command.

See docs here https://yazi-rs.github.io/docs/plugins/overview#previewer and https://yazi-rs.github.io/docs/configuration/yazi#previewers

**Reply by d-air1** [reply]
Ok thanks for the clarification and info.

**Comment by sxyazi** [comment]
I've modified the title of this discussion so that others facing the same issue can find it more easily.

BTW, if you've created a JPEG-XL previewer and are interested/willing to share it, feel free to ping me anytime, and I'll add it to the [resources page](https://yazi-rs.github.io/docs/resources) :)
